### PR TITLE
feat: added e2e tests

### PR DESCRIPTION
**Pull request by jhellar**
## Motivation

https://issues.jboss.org/browse/AGMS-328

This should help with rerunning E2E tests in future.

## Verification steps

Follow steps in README

### CODE REVIEW

File: e2e/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+For local server:
+
+1. in other terminal: `npm run server`
+2. have BrowserStackLocal running (https://www.browserstack.com/local-testing#command-line)
+
+Then:
+
+1. `npm test`
+2. test will print out `sync time`
```
Comment (wtrocki) [inline e2e/README.md]: 🤣

File: e2e/fixtures/mobile-services.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "version": 1,
+  "namespace": "perf",
+  "clientId": "perf",
+  "services": [
+    {
+      "id": "81f67bae-7d40-11e9-afde-06799ee5f0b0",
+      "name": "sync-app-perf-perf",
+      "type": "sync-app",
+      "url": "http://bs-local.com:4000/graphql",
```
Comment (wtrocki) [inline e2e/fixtures/mobile-services.json]: ??

Comment (wtrocki) [inline e2e/fixtures/mobile-services.json]: Really clever

**Comment by psturc** [comment]
👀

File: e2e/src/test.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  console.log('connecting to client');
+
+  const client = await wdio.remote(opts);
+
+  console.log('waitng for initialization');
+
+  await new Promise(resolve => setTimeout(resolve, 10000));
+
+  if (keycloakEnabled()) {
+    await authenticate(cleint);
```
Comment (psturc) [inline e2e/src/test.js]:
```suggestion
    await authenticate(client);
```

File: e2e/src/test.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  const mainWindow = await client.getWindowHandle();
+  const allWindows = await client.getWindowHandles();
+  const loginPage = allWindows.find(w => w !== mainWindow);
+  await client.switchToWindow(loginPage);
+
+  const usernamEl = await client.$('#username')
+  await usernamEl.setValue('test');
+  
+  const passwordEl = await client.$('#password')
+  await passwordEl.setValue('123');
```
Comment (psturc) [inline e2e/src/test.js]: wdyt about taking the values for keycloak user&pass from env vars and documenting it?

File: e2e/src/network-log.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+const timings = create.reduce((prev, curr) => {
+  return {
+    time: prev.time + curr.time,
+    send: prev.send + curr.timings.send,
+    receive: prev.receive + curr.timings.receive,
+    wait: prev.wait + curr.timings.wait,
+  }
+}, { time: 0, send: 0, receive: 0, wait: 0 })
+
+console.log('netowrk timings in miliseconds:', timings);
```
Comment (psturc) [inline e2e/src/network-log.js]:
```suggestion
console.log('network timings in miliseconds:', timings);
```

File: e2e/src/test.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+  
+  const loginEl = await client.$('#kc-login')
+  await loginEl.click();
+
+  await client.switchToWindow(mainWindow);
+
+  console.log('waitng for initialization');
```
Comment (psturc) [inline e2e/src/test.js]:
```suggestion
  console.log('waiting for initialization');
```

File: e2e/src/test.js [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+  await new Promise(resolve => setTimeout(resolve, 10000));
+};
+
+const test = async () => {
+  console.log('connecting to client');
+
+  const client = await wdio.remote(opts);
+
+  console.log('waitng for initialization');
```
Comment (psturc) [inline e2e/src/test.js]:
```suggestion
  console.log('waiting for initialization');
```

File: e2e/fixtures/mobile-services.json (hunk 2 above)
Comment (jhellar) [reply e2e/fixtures/mobile-services.json]: This is BrowserStack feature, to reach your localhost from device.

File: e2e/src/test.js (hunk 4 above)
Comment (jhellar) [reply e2e/src/test.js]: good point, updated

**Comment by jhellar** [comment]
Thanks for review, all should be fixed now.

**Pull request merged**